extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -65,49 +65,65 @@ object PrivilegesBuilder {
 
     def mergeProjection(table: Table, plan: LogicalPlan): Unit = {
       if (projectionList.isEmpty) {
-        privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        plan match {
+          case pvm: PermanentViewMarker if pvm.output.isEmpty =>
+            privilegeObjects += PrivilegeObject(table, pvm.catalogTable.schema.map(_.name))
+          case _ =>
+            privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        }
```

Comment:
Since we don't need handle Subquery, and w'd better handle view's output here to make it can merge projection.